### PR TITLE
dojson: update convertion for IdRef links

### PR DESCRIPTION
**Pull request by rerowep**
## Why are you opening this PR?

- Which task/US does it implement?
- Which issue does it fix?

## Dependencies

My PR depends on the following `rero-ils-ui`'s PR(s):

* rero/rero-ils-ui#<xx>

## How to test?

- What command should I have to run to test your PR?
- What should I test through the UI?

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?

### CODE REVIEW

**Review by iGormilhit — CHANGES_REQUESTED** [review]
Commit message proposition:

```
dojson: updates conversion for IdRef links

* Adapts the conversion of document with contribution references from
  IdRef.
* Corrects MARC to JSON transformation for subjects.

Co-Authored-by: Peter Weber <peter.weber@rero.ch>
```

File: rero_ils/modules/documents/dojson/contrib/marc21tojson/model.py [hunk 1]
```diff
@@ -82,46 +82,25 @@ def get_contribution_link(bibid, reroid, id, key, value):
...
+                return url.replace(test_host, prod_host)
+            else:
+                error_print('WARNING MEF IDREF NOT FOUND:', bibid, reroid, url,
+                            request.status_code)
+                return None
     except Exception as err:
-        error_print('WARNING NOT MEF REF:', bibid, id, key, value, err)
-    return mef_link
+        pass
+        # error_print('WARNING NOT MEF REF:', bibid, id, key, value, err)
```
Comment (BadrAly) [inline rero_ils/modules/documents/dojson/contrib/marc21tojson/model.py:103]: delete?

**Review by BadrAly — CHANGES_REQUESTED** [review]

**Review by BadrAly — APPROVED** [review]

**Review by iGormilhit — APPROVED** [review]
Commit message approved.

**Review by zannkukai — APPROVED** [review]

**Review by lauren-d — APPROVED** [review]

File: rero_ils/modules/documents/dojson/contrib/marc21tojson/model.py [hunk 2]
```diff
@@ -82,46 +82,24 @@ def get_contribution_link(bibid, reroid, id, key, value):
...
+            if request.status_code == requests.codes.ok:
+                return url.replace(test_host, prod_host)
+            else:
+                error_print('WARNING MEF IDREF NOT FOUND:', bibid, reroid, url,
+                            request.status_code)
+                return None
     except Exception as err:
-        error_print('WARNING NOT MEF REF:', bibid, id, key, value, err)
-    return mef_link
+        pass
```
Comment (jma) [inline rero_ils/modules/documents/dojson/contrib/marc21tojson/model.py:102]: Do you need some logging here?

**Review by jma — APPROVED** [review]

File: rero_ils/modules/documents/dojson/contrib/marc21tojson/model.py (hunk 2 above)
Comment (rerowep) [reply rero_ils/modules/documents/dojson/contrib/marc21tojson/model.py:102]: No, I don't think so. We only like to get id's prefixes with `(??????)`, in the moment we have only `(IDREF)` (see line 89).

**Review by BadrAly — APPROVED** [review]

**Review by iGormilhit — APPROVED** [review]
Commit message approved.

File: rero_ils/modules/documents/dojson/contrib/marc21tojson/model.py [hunk 3]
```diff
@@ -82,46 +84,23 @@ def get_contribution_link(bibid, reroid, id, key, value):
...
-            error_print('ERROR MEF REQUEST:', bibid, reroid, url,
-                        request.status_code)
 
-    except Exception as err:
-        error_print('WARNING NOT MEF REF:', bibid, id, key, value, err)
-    return mef_link
+    match = IDREF_REF_REGEX.search(id)
+    if match:
+        pid = match.group(1)
+        if key[:3] in ('100', '600', '610', '611', '700', '710', '711'):
```
Comment (zannkukai) [inline rero_ils/modules/documents/dojson/contrib/marc21tojson/model.py:91]: COSMETIC : Using parenthesis, you define a `tuple`. I think you would define a list instead. But the `in` statement works also with tuple. It's less human readable

**Review by zannkukai — APPROVED** [review]

**Review by lauren-d — APPROVED** [review]

**Review by jma — APPROVED** [review]

File: rero_ils/modules/documents/dojson/contrib/marc21tojson/model.py (hunk 3 above)
Comment (jma) [reply rero_ils/modules/documents/dojson/contrib/marc21tojson/model.py]: > COSMETIC : Using parenthesis, you define a `tuple`. I think you would define a list instead. But the `in` statement works also with tuple. It's less human readable

In pythonic way here should be a tuple as it is not modified.

**Pull request merged**